---
title: "Quality Control of the athletes plus controls MXL genotypes"
author: "R&D MiADN Department"
date: "June 24, 2019"
output:
  html_document: default
  pdf_document: default
---

This report shows quality control of the genotypes for athletes plus MXL samples from 1000genomes HapMap project.


### **1. Heterozygosity vs missing SNPs**

First, heterozigosity and missing SNP information are calculated in order to find outliers that could suggests sample genotyping errors.

**INPUT**: data/qc_miADN_plus_mxn/miADN_plus_mxn_updated_sex

Heterozygosity score calculation.

```
./inst/plink 
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_updated_sex \
--het \
--out data/qc_miADN_plus_mxn/miADN_plus_mxn
```

Perform missing genotypes.

```
./inst/plink \
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_updated_sex \
--missing \
--out data/qc_miADN_plus_mxn/miADN_plus_mxn
```

**OUTPUT**: 

* miADN_plus_mxn.imiss - missing genotypes ratio. 
* miADN_plus_mxn.het - heterozigosity in each sample.

NOTE: plot heterozygosity versus missing genotypes using the heterozygous_missing_check.R script.

```{r echo=FALSE, warning=FALSE, message=FALSE}
## Create het vs missing snps plots
setwd("~/scripts/gwas/sc/")
library(dplyr)
library(ggplot2)

## Read heterozigous check
het <- read.table("../data/qc_miADN_plus_mxn/miADN_plus_mxn.het",
                  fill=TRUE, sep="", header = TRUE)

## Read snp missing data
missing <- read.table("../data/qc_miADN_plus_mxn/miADN_plus_mxn.imiss",
                      fill=TRUE, sep="", header = TRUE)

annotations <- read.table("../data/annotations/miADN_plus_mxn_annotated_list.txt", 
                          sep="\t", 
                          stringsAsFactors = FALSE,
                          header = TRUE)
#head(missing)

## merge both checks
check <- merge(het, missing)
check <- merge(check, annotations)
#head(check)
#dim(check)
names(check) <- gsub("\\.", "", names(check))
check <- mutate(check, HET=(NNM-OHOM)/NNM)
#head(check)

## plot het vs missing snps
theme_set(theme_light())
#pdf("../figures/heterozygous_vs_missing.pdf")
ggplot(data=check, aes(x=F_MISS, y=HET, colour=ETNIA)) + 
        geom_point() + 
        geom_vline(xintercept = 0.35, color="red", 
                   linetype="dotted", size=0.8) +
        geom_hline(yintercept = 0.2, color="red", 
                   linetype="dotted", size=0.8) +
        xlab("Missing genotyping") +
        ylab("Heterozygous frequency") +
        theme(axis.title.x = element_text(size = 16, face="bold"),
              axis.title.y = element_text(size=16, face = "bold"))
#dev.off()
```

It can be seen, that control MXN genotypes as retrieved from 1000genomes HapMap project has high missing genotyping values. In order to preserve as long as possible sample controls QC must be focused on SNP filtering to remove SNP with low genotyping quality. Therefore, 0.20  and 0.46 can be selected thresholds for heterozygosity frequency and missing genotyping, respectively. 


### **2. Sex check** 

**INPUT**: 

* mestizo (bed/bim/fam)

We implemented sex check imputation. 

```
./inst/plink 
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_updated_sex 
--impute-sex 
--make-bed 
--out data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex
```

There were 146 sex discrepancies in mestizo samples and 134 after sex imputation. Create a file fail-sex-check.txt containing IDs for samples that does not pass sex-check after imputation.

```
## sex check
./inst/plink 
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex 
--check-sex 
--out data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex

## list of samples that fails passing sex check
grep PROBLEM data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex.sexcheck \
| cut -f1,2 \
> data/qc_miADN_plus_mxn/fail-sex-check.txt
```

**OUTPUT**: 

* miADN_plus_mxn_imputed_sex (bed/bim/fam) files.

* miADN_plus_mxn_imputed_sex.sexcheck - contains a report with sex discrepancies as inferred from homozygocity scores. 

* fail-sex-check.txt - contains a list of sample IDs that do not pass the QC.


### **3. Identity by state calculation**

**INPUT**: miADN_plus_mxn_imputed_sex (bed/bim/fam)

Removing high LD regions for IBS calculation.

```
./inst/plink \
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex \
--exclude data/qc_miADN_plus_mxn/high-LD-regions.txt \
--indep-pairwise 50 5 0.2 \
--out data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex
```

Perform IBS calculation.

```
### IBS calculation
./inst/plink \
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex \
--extract data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex.prune.in \
--genome \
--out data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex

### IBS pca plot
./inst/plink --bfile data/qc_miADN_plus_mxn/mestizo_imp_sex --read-genome data/qc_miADN_plus_mxn/mestizo_imp_sex.genome --cluster cc --ppc 1e-3 --mds-plot 2 --out data/qc_miADN_plus_mxn/strat1
```

The next plot shows the results of the previous PCA results.

```{r echo=FALSE, message=FALSE, warning=FALSE}
ibs <- read.table("../data/qc_miADN_plus_mxn/strat1.mds",
                  header = TRUE, fill = TRUE,
                  stringsAsFactors = FALSE)

ann_ibs <- merge(ibs, annotations)

ggplot(ann_ibs, aes(x=C1, y=C2, colour=ETNIA)) +
        geom_point() +
        geom_vline(xintercept = -0.01, color="red", 
                   linetype="dotted", size=0.8) +
        geom_hline(yintercept = 0.03, color="red", 
                   linetype="dotted", size=0.8) 
```

From this plot it can be seen again that control genotypes from 1000genomes are clustered together at higher IBS values in the PCA. This further supports removal of genotypes above -0.01 and 0.03 for C1 and C2 components, respectively.  

### **4. Ancestry Divergency**

INPUT: 

* miADN_plus_mxn_imputed_sex
* miADN_plus_mxn_imputed_sex.prune.in

Next, we perform a PCA in order to observe sample clusters and outliers as follows:

```
./inst/plink 
--bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex 
--pca 
--extract data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex.prune.in 
--out data/qc_miADN_plus_mxn/miADN_plus_mxn_PCA

```

The results are shown in the next plot.

```{r echo=FALSE, message=FALSE, warning=FALSE}
pca <- read.table("../data/qc_miADN_plus_mxn/miADN_plus_mxn_PCA.eigenvec",
                  header = FALSE, stringsAsFactors = FALSE)
names(pca)[1:2] <- c("FID", "IID") 
pca_ann <- merge(pca, select(annotations, -(V3:ID)))

ggplot(pca_ann, aes(x=V3, y=V4, colour=ETNIA)) +
        geom_point() +
        geom_vline(xintercept=0.015, col="red", 
              linetype="dotted", size=0.8) +
        geom_hline(yintercept=0.15, col="red", 
              linetype="dotted", size=0.8) 
```

Genotyped samples and MXL controls from 1000genomes hapmap form two separated clusters. Threshold to filter this samples out can be taken as 0.015 and 0.5 for C1 and C2 principal components.

**OUTPUT**: 

* miADN_plus_mxn_PCA.eigenvec


### 5. Removing samples that do not pass quality control


**INPUT**:

* miADN_plus_mxn_imputed_sex
* pass_qc_samples.txt

Using the above defined QC thresholds all the samples IDs that did not pass the quality control were stored in the table fail_qc_samples.txt using the script fail_samples_qc_R. Then, samples were filtered out excluding these samples.

```
./inst/plink --bfile data/qc_miADN_plus_mxn/miADN_plus_mxn_imputed_sex --keep data/qc_miADN_plus_mxn/pass_qc_samples.txt --make-bed --out data/qc_miADN_plus_mxn/miADN_pass_qc
```

After removing all samples that did not pass the quality control only 140 genotypes remain. 

**OUPUT**:

* miADN_pass_qc (bed/bim/fam) files

### 6. Filtering markers with low genotyping rate

**INPUT**: 

* miADN_pass_qc

Perform quality assessment of genotyping rates for markers.

```
./inst/plink --bfile data/qc_miADN_plus_mxn/miADN_pass_qc --missing --out data/qc_miADN_plus_mxn/snps-low-qual
```

The output is stored in the snps-low-qual.lmiss file. 

```{r echo=FALSE, warning=FALSE, message=FALSE}
#low_qual <- read.table("../data/qc_miADN_plus_mxn/snps-low-qual.lmiss",
#                       stringsAsFactors = FALSE,
#                       header = TRUE,
#                       fill = TRUE)

#jpeg("../figures/snps_low_qual_genotyping.jpg")
#hist(low_qual$F_MISS, breaks = 50)
#abline(v=0.05, col="red")
#dev.off()
#sum(low_qual$F_MISS<0.05)

#lq <- filter(low_qual, F_MISS > 0.05) %>%
#        select(SNP)
#write.table(lq, "../data/qc_miADN_plus_mxn/snps-low-qual.txt",
#            row.names = FALSE,
#            col.names = FALSE,
#            quote = FALSE)
```
![Low quality markers](../figures/snps_low_qual_genotyping.jpg)

We choose a genotyping threshold of 0.05. There are 1,359,945 markers that pass the quality threshold. Markers that do not pass the qc were stored in the file snps-low-qual.txt. Then, they were filtered out using the following command.

```
./inst/plink 
--bfile data/qc_miADN_plus_mxn/miADN_pass_qc 
--exclude data/qc_miADN_plus_mxn/snps-low-qual.txt 
--make-bed 
--out data/qc_miADN_plus_mxn/miADN_markers_miss_qc
```

**OUTPUT**:

* miADN_markers_miss_qc (bed/bim/fam)

**Conclusions**: Quality control of the sample genotypes reveals by using multiple methodologies (Heterozygosity vs missing data plot, IBS and PCA) that MXL controls are clustered apart from our data. This suggests that MXL samples from 1000genomes could not be good controls for GWAs analysis of the endurance athletes since it may cause bias related to population stratification.  